src/java.desktop/share/classes/javax/swing/JPopupMenu.java

```diff
@@ -929,6 +931,16 @@ public Component getInvoker() {
         return this.invoker;
     }
 
+    private class Listener implements PropertyChangeListener,Serializable {
```

Comment:
```suggestion
    private final class AncestorListener implements PropertyChangeListener, Serializable {
```
Be more specific. Make the class `final` explicitly; it's not meant to be extended. Put a space after the comma.

Reply:
ok